package/MDAnalysis/core/_get_readers.py

```diff
@@ -83,20 +84,15 @@ def get_reader_for(filename, format=None):
         return format
 
     # ChainReader gets returned even if format is specified
-    if not isinstance(filename, np.ndarray) and util.iterable(filename) and not isstream(filename):
+    if _READER_HINTS['CHAIN'](filename):
         format = 'CHAIN'
     # Only guess if format is not specified
-    elif format is None:
-        # Checks for specialised formats
-        if isinstance(filename, np.ndarray):
-            # memoryreader slurps numpy arrays
-            format = 'MEMORY'
-        elif isinstance(filename, mmtf.MMTFDecoder):
-            # mmtf slurps mmtf object
-            format = 'MMTF'
-        elif isinstance(filename, pmd.Structure):
-            format = 'PARMED'
-        else:
+    if format is None:
+        for f, test in _READER_HINTS.items():
+            if test(filename):
+                format = f
+                break;
```

Comment:
Don't need the semicolon

Reply:
haha I've been doing too little Python recently :D